### PR TITLE
SIGINT-2052 Bump up @actions/artifact to 2.1.8 to use v4 upload API

### PR DESCRIPTION
**Pull request by sadmananik**

### CODE REVIEW

**Comment by sadmananik** [comment]
@kishorikumar I am not sure why check dist is showing diff. I did not find any change in local. Can you please check this once?

File: package.json [hunk 1]
```diff
@@ -45,7 +45,7 @@
     "typescript": "^4.7.4"
   },
   "dependencies": {
-    "@actions/artifact": "^1.1.0",
+    "@actions/artifact": "^2.1.8",
```
Comment (kishorikumar) [inline package.json:48]: As per the npmjs release page (https://www.npmjs.com/package/@actions/artifact), this version might not be supported GHES. 
Test once, if artifact library is supported on GHES

**Review by spurohitsynopsys — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (sadmananik) [reply package.json]: Resolved. I had to use two diff versions to work on both GHC and GHE.

File: src/synopsys-action/artifacts.ts [hunk 2]
```diff
@@ -51,11 +65,20 @@ export function getFiles(dir: string, allFiles: string[]): string[] {
   return allFiles
 }
 
-export async function uploadSarifReportAsArtifact(defaultSarifReportDirectory: string, userSarifFilePath: string, artifactName: string): Promise<UploadResponse> {
-  const artifactClient = artifact.create()
+export async function uploadSarifReportAsArtifact(defaultSarifReportDirectory: string, userSarifFilePath: string, artifactName: string): Promise<UploadArtifactResponse> {
+  const githubServerUrl = process.env[constants.GITHUB_ENVIRONMENT_VARIABLES.GITHUB_SERVER_URL] || ''
+  const isGitHubCloud: boolean = githubServerUrl === constants.GITHUB_CLOUD_URL
```
Comment (kishorikumar) [inline src/synopsys-action/artifacts.ts:70]: As this check is happening at multiple places, we can have this check at one place (may be in utility) to reduce redundancy.

**Review by kishorikumar — APPROVED** [review]

File: src/synopsys-action/artifacts.ts (hunk 2 above)
Comment (sadmananik) [reply src/synopsys-action/artifacts.ts]: Resolved

**Pull request merged**